src/transformers/modeling_utils.py

```diff
@@ -1933,15 +1933,22 @@ def save_pretrained(
         if token is not None:
             kwargs["token"] = token
 
+        _hf_peft_config_loaded = getattr(self, "_hf_peft_config_loaded", False)
+
         # Checks if the model has been loaded in 8-bit
-        if getattr(self, "is_loaded_in_8bit", False) and getattr(self, "is_8bit_serializable", False):
+        if (
+            getattr(self, "is_loaded_in_8bit", False)
+            and getattr(self, "is_8bit_serializable", False)
+            and not _hf_peft_config_loaded
+        ):
             warnings.warn(
                 "You are calling `save_pretrained` to a 8-bit converted model you may likely encounter unexepected"
                 " behaviors. If you want to save 8-bit models, make sure to have `bitsandbytes>0.37.2` installed.",
                 UserWarning,
             )
```

Comment:
Should we check if there is bitsandbytes>0.37.2 installed and raise an error instead of sending out a warning?